### ISSUE TITLE
Transactions table ignores pagination (returns all records)

### ISSUE THREAD

**Issue by syrus83** [malsubrata/woo-wallet#97]
On the My Wallet “Transactions” tab, the DataTables request is server-side, but the response always contains all transactions. Pagination and per-page limits are ignored.

Steps to Reproduce:

Log in and open My Account → My Wallet → Transactions tab.
Observe the table shows all transactions and pagination controls don’t change the result.
Check the network request to admin-ajax.php with action=draw_wallet_transaction_details_table — note it sends start/length and search params.
Expected: Server applies LIMIT offset,rows and returns only the current page. Pagination works.

Actual: Server returns the full transaction list; pagination has no effect.

Technical Details:

JS sends the limit as "{$start}, {$length}" with a space after the comma.
draw_wallet_transaction_details_table() builds $args['limit'] = "$start, $length"; @includes/class-woo-wallet-ajax.php#396-419.
get_wallet_transactions() only applies LIMIT when the string matches /^\d+,\d+$/ (no spaces) @includes/helper/woo-wallet-util.php#333-501. Because of the space, the regex fails and LIMIT is omitted, so all rows are returned.
Suggested Fix:

Remove the space when building the limit string (e.g., "$start,$length"), or trim/normalize before passing to get_wallet_transactions().

for others who want a workable solution without touching the plugin core files  paste the below in either your theme functions.php or any code snippet of your choice  

`// Normalize woo wallet transaction limits to enforce pagination.
add_filter( 'woo_wallet_transactions_query_args', function ( $args ) {
	if ( ! empty( $args['limit'] ) && is_string( $args['limit'] ) ) {
		// Remove spaces around comma so it matches the expected "offset,limit" pattern.
		$args['limit'] = preg_replace( '/\s*,\s*/', ',', $args['limit'] );
	}
	return $args;
} );`
